### DISCUSSION TITLE
Technical architecture discussion

### DISCUSSION THREAD

**Discussion by kljensen** [yale-research-job-board/.github#2] [General]
Just some quick thoughts.

* This will be a simple app running on a single server.
* I'm thinking this will be the best learning experience for folks if we run it on a VM of our own control rather than a PAAS. Likely we'll end up running on Yale's Spin Up service and punching a hole through the firewall with [cloudflared](https://github.com/cloudflare/cloudflared).
* We will use [terraform](https://www.terraform.io) and [ansible](https://github.com/ansible/ansible) for provisioning.
* Likely we'll just run everything with [docker compose](https://docs.docker.com/compose/) in a persistent [tmux](https://tmuxcheatsheet.com) session to start (pretty non-professional but likely fine as we flesh things out.
* We want something that Tristan can hand to other developers, so good to pick a well-known framework and language like python's [django](https://www.djangoproject.com). I'm set on this rather than Flask or FastAPI for all the goodies he'd get for free, like the admin panel.
* I find the Django migrations distasteful (leaky abstraction). So, I'd prefer to write SQL migrations by hand and make Django models "unmanaged".
* I can't decide between SQLite and Postgres. I suspect the former is the right choice for this small project.
* We'll stick [Caddy](https://caddyserver.com) in front of the app as a reverse proxy.
* We'll keep client-side JS to a minimum, perhaps using [htmx](https://htmx.org) where we need something fancy.
* I'm keen to use a classless CSS framework, but open to alternatives. I don't want to get too into the design aspect. Hoping to get an MVP running and hand it off.
* We will use [uv](https://github.com/astral-sh/uv) for package management in python.

I'd be grateful for discussion and debate.

**Comment by kljensen** [comment]
Likely Tristan will want to do some fancy analytics stuff, so we might think about adding [Plausible](https://github.com/plausible/analytics) or [Umami](https://github.com/umami-software/umami). I dunno. This is likely second order.

**Comment by kljensen** [comment]
Might add [pgcat](https://github.com/postgresml/pgcat) connection pooler. Not much confidence in Django pooler and seems trivial to add with Docker.

**Comment by fraxinus1098** [comment]
@yashswijain @dikshita-gn @yiyanghe1 @komalbagrodia97 regarding your questions on the project architecture for our sync with Kyle:
1. What are you hoping to get from the meeting that you couldn't get from Google/YouTube/ChatGPT, etc
2. What specific questions do you have? What do you already understand vs need more clarification on?
3. What is the proposed meeting agenda?

**Comment by fraxinus1098** [comment]
Existing job board competitive analysis https://yaleedu-my.sharepoint.com/:x:/r/personal/ash_duong_yale_edu/_layouts/15/Doc.aspx?sourcedoc=%7B9E170293-0DF8-4B18-A659-F240500CA706%7D&file=Initial%20Survey.xlsx&action=default&mobileredirect=true

**Comment by fraxinus1098** [comment]
Pull jobs from https://venturewell.org/
